ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs

```diff
@@ -6,156 +6,112 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Rendering;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.EntityFrameworkCore.Metadata.Internal;
 using ParkingZoneApp.Data;
 using ParkingZoneApp.Models;
+using ParkingZoneApp.Repositories;
 
-namespace ParkingZoneApp.Areas.Admin.Controllers
+namespace ParkingZoneApp.Areas.Admin
 {
     [Authorize]
     [Area("Admin")]
     public class ParkingZoneController : Controller
     {
-        private readonly ApplicationDbContext _context;
+        private readonly IParkingZoneRepository _repository;
 
-        public ParkingZoneController(ApplicationDbContext context)
+        public ParkingZoneController(IParkingZoneRepository repository)
         {
-            _context = context;
+            _repository = repository;
         }
-        // GET: ParkingZone
-        
-        public async Task<IActionResult> Index()
+
+        // GET: Admin/ParkingZone
+        public IActionResult Index()
         {
-            return View(await _context.ParkingZone.ToListAsync());
+            var AllParkingZone = _repository.GetAll().ToList();
+            return View(AllParkingZone);
         }
 
-        // GET: ParkingZone/Details/5
-        public async Task<IActionResult> Details(int? id)
+        // GET: Admin/ParkingZone/Details/5
+        public IActionResult Details(int id)
         {
-            if (id == null)
-            {
-                return NotFound();
-            }
-
-            var parkingZoneModel = await _context.ParkingZone
-                .FirstOrDefaultAsync(m => m.Id == id);
-            if (parkingZoneModel == null)
+            var ParkingZoneById = _repository.GetById(id);
+            if (ParkingZoneById is not null)
```

Comment:
you checked ParkingZoneById for null its good. What if id is null? Details(int id) add if for it